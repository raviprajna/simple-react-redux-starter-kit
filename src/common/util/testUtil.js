import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export function getComponentTree( component){
    const wrapper = shallow(component);
    const shallowComponent = wrapper.dive();
    return toJson(shallowComponent);
}
export function getWrapper( component){
    const wrapper = shallow(component);
    return wrapper;
}