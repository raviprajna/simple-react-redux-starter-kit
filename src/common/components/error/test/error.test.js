import React from 'react';
import Error from '../';
import {getComponentTree} from '../../../util/testUtil.js'

describe('<Error /> Component Test', () => {

  it('renders <Error /> no props component', () => {
    expect(getComponentTree(<Error />)).toMatchSnapshot();
  });

  it('renders <Error /> component with error object as props ', () => {

    expect(getComponentTree(
    <Error error={{'message' : 'Error !!'}}/>
    )).toMatchSnapshot();

  });

});