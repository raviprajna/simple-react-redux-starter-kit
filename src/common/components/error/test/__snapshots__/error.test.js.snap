// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> Component Test renders <Error /> component with error object as props  1`] = `
<StyledComponent
  color="red"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
      },
      "displayName": "Styled(styled.div)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "propTypes": Object {
        "fontFamily": [Function],
        "fontWeight": [Function],
        "letterSpacing": [Function],
        "lineHeight": [Function],
        "textAlign": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Error !!
</StyledComponent>
`;

exports[`<Error /> Component Test renders <Error /> no props component 1`] = `
<StyledComponent
  color="red"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
      },
      "displayName": "Styled(styled.div)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "propTypes": Object {
        "fontFamily": [Function],
        "fontWeight": [Function],
        "letterSpacing": [Function],
        "lineHeight": [Function],
        "textAlign": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Something went wrong
</StyledComponent>
`;
